### DISCUSSION TITLE
Adding `minIssuedAt` / `minTokenDate` to `JWTVerifyOptions`?

### DISCUSSION THREAD

**Discussion by Manouchehri** [panva/jose#576] [Ideas]
Example usage:

```typescript
const jwt_verified = await jwtVerify(unchecked_jwt_string, pubkey, {
      algorithms: ['ES256'],
      issuer: 'panva.demo.ai.moda',
      audience: 'panva',
      maxTokenAge: '31536000s',
      minIssuedAt: new Date(1672531200000),
      requiredClaims: ['sub', 'jti']
    })
```

And it would check that the `iat` is past the `minIssuedAt` date.

**Comment by panva** [comment]
I think you should look at the `nbf` claim.

**Reply by Manouchehri** [reply]
Ah yep, so I basically want to verify a "`nbf`" date with `jwtVerify`, without actually needing the `nbf` present on the signed token. =) The existing tokens already have `iat` on them.

**Reply by panva** [reply]
> it would check that the iat is past the minIssuedAt date

Hey, but that's what `maxTokenAge` already does, only not through passing of a Date but rather through elapsed time.

**Reply by Manouchehri** [reply]
Good point! Think we could add support for passing a Date object to `maxTokenAge`?

**Reply by panva** [reply]
Sorry, I don't see the point. If you know the date you want as minimum then either use the existing maxTokenAge option of passing the seconds elapsed since, or simply evaluate the `iat` from the verified payload afterwards.

**Reply by Manouchehri** [reply]
```typescript
const jwt_verified = await jwtVerify(unchecked_jwt_string, pubkey, {
      algorithms: ['ES256'],
      issuer: 'panva.demo.ai.moda',
      audience: 'panva',
      maxTokenAge: Math.min(Math.floor((Date.now() - new Date("Dec 31 2022 19:00:00").getTime())/1000), 31536000),
      minIssuedAt: new Date(1672531200000),
      requiredClaims: ['sub', 'jti']
    })
```

True, it feels a bit messy though?

**Reply by panva** [reply]
If you insist on inlining it, sure, feels ugly. But functional.

**Reply by Manouchehri** [reply]
If I make a PR for `maxTokenAge` to accept a `Date`, will you consider merging it? =)

**Reply by panva** [reply]
I will not, the name `maxTokenAge` does not lend itself semantically to accepting a Date object.
